### PR TITLE
[DEVOPS-1393] Added dialog to confirm filter output and integrations tests

### PR DESCRIPTION
**Pull request by AndreKrebs**

### CODE REVIEW

File: core/src/main/resources/static/scripts/drawer-panel-service.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    register(paperDrawerPanel) {
+        this.paperDrawerPanel = paperDrawerPanel;
+    }
+
+    drawerPannelToggle() {
+        var drawerSelected = this.paperDrawerPanel.getAttribute("selected");
+
+        if (drawerSelected === "main")
+            this._selected();
+        else
```
Comment (objective-solutions-user) [inline core/src/main/resources/static/scripts/drawer-panel-service.js]: MINOR Remove the useless trailing whitespaces at the end of this line. [rule](http://10.44.1.250:9000/coding_rules#rule_key=javascript%3ATrailingWhitespace)

**Comment by objective-solutions-user** [comment]
SonarQube analysis reported 1 issue
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: application/src/test/java/objective/taskboard/it/BoardFilterIT.java [hunk 2]
```diff
@@ -123,4 +129,93 @@ public void allFilterFeatures() {
                     "TASKB-641"
                 );
     }
+
+    @Test
+    public void whenClickScrim_checkModal() {
+        // Test "click outside after revert"
```
Comment (mauricio-frontend) [inline application/src/test/java/objective/taskboard/it/BoardFilterIT.java]: You can remove this comment.

File: application/src/test/java/objective/taskboard/it/BoardFilterIT.java [hunk 3]
```diff
@@ -123,4 +129,93 @@ public void allFilterFeatures() {
...
+                .openMenuFilters()
+                .openCardFieldFilters()
+                .clickCheckAllFilter("Project")
+                .clickButtonRevertFilter()
+                .clickOutsideFilter()
+                .assertNotExistsModalDiscardFilter();
+
+        mainPage.reload();
+
+        // Test "check modal button `Confirm`"
```
Comment (mauricio-frontend) [inline application/src/test/java/objective/taskboard/it/BoardFilterIT.java]: Remove the comment.

File: application/src/test/java/objective/taskboard/it/BoardFilterIT.java [hunk 4]
```diff
@@ -123,4 +129,93 @@ public void allFilterFeatures() {
...
+                        "PROJ2 | PROJECT 2",
+                        "PROJ3 | PROJECT 3",
+                        "PROJ5 | PROJECT 5",
+                        "PROJ6 | PROJECT 6",
+                        "PROJ8 | PROJECT 8",
+                        "TASKB | TASKB-SUB-PROJ");
+
+        mainPage.reload();
+
+        // Test "check modal button `Cancel`"
```
Comment (mauricio-frontend) [inline application/src/test/java/objective/taskboard/it/BoardFilterIT.java]: Remove the comment.

File: application/src/test/java/objective/taskboard/it/BoardFilterIT.java [hunk 5]
```diff
@@ -123,4 +129,93 @@ public void allFilterFeatures() {
...
+                .clickFilterFieldValue("Issue Type", "FEATURE", "QA", "TASK")
+                .clickOutsideFilter()
+                .assertExistsModalDiscardFilter()
+                .clickCancelModalDiscardFilter()
+                .assertNotExistsModalDiscardFilter()
+                .assertBoardFilterFieldsUnmarkedOptions("FEATURE", "QA", "TASK");
+
+        mainPage.reload();
+
+        // Teste "apply filter and check not exists modal"
```
Comment (mauricio-frontend) [inline application/src/test/java/objective/taskboard/it/BoardFilterIT.java]: Remove the comment.

File: application/src/test/java/objective/taskboard/it/BoardFilterIT.java [hunk 6]
```diff
@@ -123,4 +129,93 @@ public void allFilterFeatures() {
...
+                .clickFilterFieldValue("Issue Type", "FEATURE", "QA", "TASK")
+                .clickButtonRevertFilter()
+                .assertBoardFilterFieldsUnmarkedOptions("")
+                .assertBoardFilterButtonsInitialState()
+                .clickOutsideFilter()
+                .assertNotExistsModalDiscardFilter();
+
+        mainPage.reload();
+
+        // Teste "apply filter and check not exists modal"
```
Comment (mauricio-frontend) [inline application/src/test/java/objective/taskboard/it/BoardFilterIT.java]: Remove the comment.

File: cardboard/src/main/resources/static/elements/taskboard/card-field-filters.html [hunk 7]
```diff
@@ -81,7 +81,8 @@
             }
         </style>
 
-        <iron-signals on-iron-signal-update-board-filters="_setInitialFilter"></iron-signals>
+        <iron-signals on-iron-signal-update-board-filters="_setInitialFilter"
+        on-iron></iron-signals>
```
Comment (mauricio-frontend) [inline cardboard/src/main/resources/static/elements/taskboard/card-field-filters.html]: Can keep this code in the same line, apparently this 'on-iron' here is not necessary.

Comment (AndreKrebs) [reply cardboard/src/main/resources/static/elements/taskboard/card-field-filters.html]: Ok

File: application/src/test/java/objective/taskboard/it/BoardFilterIT.java (hunk 6 above)
Comment (AndreKrebs) [reply application/src/test/java/objective/taskboard/it/BoardFilterIT.java]: Moved to new method

File: application/src/test/java/objective/taskboard/it/BoardFilterIT.java (hunk 5 above)
Comment (AndreKrebs) [reply application/src/test/java/objective/taskboard/it/BoardFilterIT.java]: Moved to new method

File: application/src/test/java/objective/taskboard/it/BoardFilterIT.java (hunk 4 above)
Comment (AndreKrebs) [reply application/src/test/java/objective/taskboard/it/BoardFilterIT.java]: Moved to new method

File: application/src/test/java/objective/taskboard/it/BoardFilterIT.java (hunk 3 above)
Comment (AndreKrebs) [reply application/src/test/java/objective/taskboard/it/BoardFilterIT.java]: Moved to new method

File: application/src/test/java/objective/taskboard/it/BoardFilterIT.java (hunk 2 above)
Comment (AndreKrebs) [reply application/src/test/java/objective/taskboard/it/BoardFilterIT.java]: Moved to new method

**Pull request merged**